src/components/MovieView.js

```diff
@@ -18,13 +18,17 @@ class MovieView extends Component {
         apiCalls.getTrailerData(this.props.selectedMovie.id)
             .then(response => checkResponse(response))
             .then(data => this.setState(prevState => {
-               return {trailer: `${prevState.trailer}${data.videos[0].key}`}
+               if (!data.videos.length) {
+                 return
+               }
+               const key = data.videos.find(video => video.type === 'Trailer').key
+               return {trailer: `${prevState.trailer}${key}`}
                 }))
```

Comment:
Smart, great catch with going through all the movies. We could have totally missed that some of the movies don't have videos.